docs/bootstrapping-a-kubernetes-cluster.md

```diff
@@ -0,0 +1,144 @@
+role.
+
+To configure the installation, create an `extra_vars.yml` file to capture the variables for your
+environment. The following is a sample that includes commonly used variables:
+
+```yaml
+etcd_interface: eth0 # Interface that etcd should bind
+kubernetes_common_primary_interface: eth0 # Interface that should be used to obtain the node's IP
+kubernetes_cni_plugin: calico # The CNI plugin to use
+```
+
+_Note: Wardroom sets Ansible's [hash merging behavior](https://docs.ansible.com/ansible/latest/reference_appendices/config.html#default-hash-behaviour)
+to `merge`. This means that any variable that is a hash (aka. map or dictionary) with a default
+value will be merged with the user-provided variable._
+
+#### Kubernetes API IP or FQDN
+
+When setting up a highly-available Kubernetes cluster, the control plane must be accessible through
+a single address. This is typically achieved by deploying a load balancer in front of the control
+plane nodes.
+
+Wardroom exposes two variables to provide the load-balanced API address:
+
+```yaml
+# IP address of the load balancer fronting the Kubernetes API servers.
+# This is required if an FQDN is not provided. Otherwise, this can be left empty.
+kubernetes_common_api_ip: ""
+
+# FQDN of the load balancer fronting the Kubernetes API servers.
+# This variable takes precedence over the kubernetes_common_api_ip variable.
+kubernetes_common_api_fqdn: ""
+```
+
+#### Modifying Kubeadm Configuration
+
+The kubeadm configuration is provided via the `kubernetes_common_kubeadm_config` variable. Wardroom
+provides a [default](../ansible/roles/kubernetes-common/defaults/main.yml) for this variable that
+can be modified through the `extra_vars.yml` file.
+
+Given that the variable is a hash (or map), the user-provided variable is merged with the default
+variable. This provides the ability to make point modifications without having to redefine the
+entire variable.
+
+The following example shows how to set the pod CIDR in the `extra_vars.yml` file:
+
+```yaml
+kubernetes_common_kubeadm_config:
+  networking:
+    podSubnet: "172.16.0.0/16"
+```
+
+#### Modifying CNI Manifests
+
+Wardroom provides a mechanism to modify the YAML deployment manfiests of the CNI plugin before
+installing it in the cluster.
+
+The following example shows how to set the `CALICO_IPV4POOL_CIDR` environment variable in the
+Calico deployment manifests. This must be defined in the `extra_vars.yml` file:
+
+```yaml
+kubernetes_cni_calico_manifest_mods:
+  - conditions:
+    - expression: kind
+      value: DaemonSet
+    modifications:
+    - expression: "spec.template.spec.containers[?(@.name == 'calico-node')].env[?(@.name == 'CALICO_IPV4POOL_CIDR')].value"
+      value: "172.16.0.0/16"
+```
+
+### Run Ansible
+
+Once you have the inventory and extra variables files, run the playbook:
+
+```shell
+ansible-playbook install.yml --inventory inventory.ini --extra-vars @extra_vars.yml
```

Comment:
So the "eaxmples/" directory contains inventories. Not sure how we want to express that here.

Reply:
Not sure I follow. Should we be referencing the `examples/` dir here somehow? Or is the concern about recommending the usage of an `extra_vars` file?